subsys/bluetooth/host/gatt.c

```diff
@@ -2542,20 +2531,40 @@ int bt_gatt_notify_cb(struct bt_conn *conn,
 int bt_gatt_notify_multiple(struct bt_conn *conn, uint16_t num_params,
 			    struct bt_gatt_notify_params *params)
 {
-	int i, ret;
-
 	__ASSERT(params, "invalid parameters\n");
-	__ASSERT(num_params, "invalid parameters\n");
-	__ASSERT(params->attr, "invalid parameters\n");
+	__ASSERT(num_params >= 2, "invalid parameters\n");
+	__ASSERT(params->attr || params->uuid, "invalid parameters\n");
+	for (uint16_t i = 1; i < num_params; i++) {
+		__ASSERT(params[0].func == params[i].func, "invalid parameters\n");
+		__ASSERT(memcmp(params[0].user_data, params[i].user_data, 1) == 0,
+			"invalid parameters\n");
+	}
 
-	for (i = 0; i < num_params; i++) {
-		ret = bt_gatt_notify_cb(conn, &params[i]);
-		if (ret < 0) {
-			return ret;
-		}
+	if (!atomic_test_bit(bt_dev.flags, BT_DEV_READY)) {
+		return -EAGAIN;
 	}
 
-	return 0;
+	if (conn && conn->state != BT_CONN_CONNECTED) {
+		return -ENOTCONN;
+	}
+
+#if defined(CONFIG_BT_GATT_ENFORCE_CHANGE_UNAWARE)
+	/* BLUETOOTH CORE SPECIFICATION Version 5.1 | Vol 3, Part G page 2350:
+	 * Except for the Handle Value indication, the server shall not send
+	 * notifications and indications to such a client until it becomes
+	 * change-aware.
+	 */
+	if (!bt_gatt_change_aware(conn, false)) {
+		return -EAGAIN;
+	}
+#endif
+
+	if (gatt_cf_notify_multi(conn)) {
+		return gatt_notify_mult(conn, num_params, params);
+	}
+
+	BT_ERR("Multiple variable length notifications is not supported");
+	return -EOPNOTSUPP;
```

Comment:
I don't think it is necessary to log an error when we are directly returning an error code. 

```suggestion
	BT_DBG("Multiple variable length notifications is not supported");
	
	return -EOPNOTSUPP;
```

Reply:
In that case, I think we need to document this error code in the documentation of `bt_gatt_notify_multiple`. Anyway, I found the message a bit confusing. I would say "The peer does not have "Multiple Handle
Value Notifications" in its Client Supported Features."

Reply:
Or perhaps simply `BT_DBG("Multiple variable length notifications is not supported by peer");`